### PR TITLE
fixing volume calculation bug

### PR DESCRIPTION
**Pull request by seichhorn**
fixing bug in how the cell volume was calculated

### CODE REVIEW

File: merlin/util/spatialfeature.py [hunk 1]
```diff
@@ -172,17 +172,17 @@ def get_volume(self) -> float:
...
-                totalVolume += deltaZ*np.sum([x.area for x in b])
-        else:
-            totalVolume = np.sum([x.area for x in boundaries[0]])
+        zPos = np.array(self._zCoordinates)
+        zDiff = np.diff(zPos)
+        zNum = np.array([[x,x+1] for x in list(range(len(zPos)-1))])
+        areas = np.array([np.sum([y.area for y in x]) if len(x) > 0 
+            else 0 for x in boundaries])
+        totalVolume = np.sum([np.mean(areas[zNum[x]]) * zDiff[x] 
+            for x in range(zNum.shape[0])])
```
Comment (emanuega) [inline merlin/util/spatialfeature.py]: Can you cover the edge case where len(zPos) = 1? Here we can assume a height of 1 and restore the original docstring. This should fix the tests that are failing.

File: merlin/util/spatialfeature.py [hunk 2]
```diff
@@ -172,17 +172,17 @@ def get_volume(self) -> float:
...
-        totalVolume = 0
 
-        if len(boundaries) > 1:
-            for b, deltaZ in zip(boundaries[:1], np.diff(self._zCoordinates)):
-                totalVolume += deltaZ*np.sum([x.area for x in b])
-        else:
-            totalVolume = np.sum([x.area for x in boundaries[0]])
+        zPos = np.array(self._zCoordinates)
+        zDiff = np.diff(zPos)
+        zNum = np.array([[x,x+1] for x in list(range(len(zPos)-1))])
```
Comment (emanuega) [inline merlin/util/spatialfeature.py]: list is redundant

**Comment by seichhorn** [comment]
Should be all set now.
